### PR TITLE
feat: Eigen Blob ID in L1

### PR DESCRIPTION
**Pull request by gianbelinche**
## What ❔
This PR adds the blob id to be stored on L1 when a batch is committed
It also adds a script for retrieving them directly from L1.
To run it:
```
cd get_all_blobs
npm i
node getallblobs.js validatorTimelockAddress=<validatorTimelockAddress> commitBatchesSharedBridge_functionSelector=<commitBatchesSharedBridge_functionSelector>
```
This generates a `blob_data.json` file, where blobs and commitments are stored.

Check `validatorTimelockAddress` when running `zk init`
`commitBatchesSharedBridge_functionSelector` is usually `0x6edd4f12`

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

### CODE REVIEW

File: get_all_blobs/getallblobs.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  }
+
+  // Convert byte array to UTF-8 string
+  const utf8String = new TextDecoder('utf-8').decode(new Uint8Array(bytes));
+  return utf8String;
+}
+
+async function getTransactions() {
+  const latestBlock = await web3.eth.getBlockNumber();
+  const validatorTimelockAddress = "0xeacf0411de906bdd8f2576692486383797d06004";
```
Comment (juanbono) [inline get_all_blobs/getallblobs.js:150]: please parameterize the `validatorTimelockAddress` and the `commitBatchesSharedBridge_functionSelector`

File: get_all_blobs/getallblobs.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+          console.log(`Decoded Commitment:`, commitment);
+        }
+      }
+    });
+  }
+}
+
+getTransactions();
+
+//0x4ed3cbf1cf6e8738118f87e5060aee0817c6f18b Chain Admin
```
Comment (juanbono) [inline get_all_blobs/getallblobs.js:174]: what is the purpose of these comments?

**Review by juanbono — CHANGES_REQUESTED** [review]
Please address the comments. Also can you add some docs about how to run it?

File: get_all_blobs/getallblobs.js (hunk 2 above)
Comment (gianbelinche) [reply get_all_blobs/getallblobs.js:174]: They are all the contracts that are called on L1, I think it is nice to leave them somewhere is case we need them in the future. That could go somewhere else tough

File: get_all_blobs/getallblobs.js (hunk 1 above)
Comment (gianbelinche) [reply get_all_blobs/getallblobs.js]: Done!

**Comment by gianbelinche** [comment]
> Please address the comments. Also can you add some docs about how to run it?

Done!

File: get_all_blobs/getallblobs.js [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        const input = tx.input;
+        const txSelector = input.slice(0, 10);
+        if (txSelector == commitBatchesSharedBridge_functionSelector) {
+          const functionAbi = contract.options.jsonInterface.find(item => item.name === 'commitBatchesSharedBridge');
+          const decodedParams = web3.eth.abi.decodeParameters(
+            functionAbi.inputs,
+            input.slice(10) // Remove the function selector (first 10 characters of the calldata)
+          );
+          commitment = hexToUtf8(decodedParams._newBatchesData[0].pubdataCommitments.slice(4));
+          console.log(`Decoded Commitment:`, commitment);
```
Comment (juanbono) [inline get_all_blobs/getallblobs.js:163]: please save/print the blobs too.

**Review by juanbono — APPROVED** [review]

**Comment by juan518munoz** [comment]
Had some issues running the mentioned procedure
- `node install` failed -> instead I ran `npm i`
- `node getallblobs.js` threw connection refused -> replaced `web3``localhost`  instantiation with `127.0.0.1`

I would recommend applying the second change.

**Review by juan518munoz — APPROVED** [review]

**Comment by gianbelinche** [comment]
> Had some issues running the mentioned procedure
> 
> * `node install` failed -> instead I ran `npm i`

Yeah, typo, its npm

**Pull request merged**